client/header/activity-panel/index.js

```diff
@@ -191,40 +221,74 @@ export class ActivityPanel extends Component {
 	}
 
 	getPanelContent( tab ) {
+		const { hasUnreadOrders, query, hasUnapprovedReviews } = this.props;
+		const { task } = query;
+
 		switch ( tab ) {
 			case 'inbox':
 				return <InboxPanel />;
 			case 'orders':
-				const { hasUnreadOrders } = this.props;
 				return <OrdersPanel hasActionableOrders={ hasUnreadOrders } />;
 			case 'stock':
 				return <StockPanel />;
 			case 'reviews':
-				const { hasUnapprovedReviews } = this.props;
 				return (
 					<ReviewsPanel
 						hasUnapprovedReviews={ hasUnapprovedReviews }
 					/>
 				);
 			case 'help':
-				const { query } = this.props;
-				const { task } = query;
 				return <HelpPanel taskName={ task } />;
 			default:
 				return null;
 		}
 	}
 
 	renderPanel() {
+		const { updateOptions } = this.props;
 		const { isPanelOpen, currentTab, isPanelSwitching } = this.state;
-
 		const tab = find( this.getTabs(), { name: currentTab } );
+
 		if ( ! tab ) {
 			return (
 				<div className="woocommerce-layout__activity-panel-wrapper" />
 			);
 		}
 
+		const clearPanel = () => {
+			this.clearPanel();
+		};
+
+		if ( currentTab === 'setup' ) {
+			const currentLocation = window.location.href;
+			const homescreenLocation = getAdminLink(
+				'admin.php?page=wc-admin'
+			);
+
+			// Don't navigate if we're already on the homescreen, this will cause an infinite loop
+			if ( currentLocation !== homescreenLocation ) {
+				// Ensure that if the user is trying to get to the task list they can see it even if
+				// it was dismissed.
+				updateOptions( {
+					woocommerce_task_list_hidden: 'no',
+				} ).then( () => {
```

Comment:
This is a nice addition. I see below we gather `woocommerce_task_list_hidden`. Would it make sense to see if this option is already "no" and skip the update to avoid the waiting for the request?

Reply:
Yeah good idea. Will fix.